### PR TITLE
Included auc or mean aggregation to faithfulness metrics.

### PR DESCRIPTION
**Pull request by Wickstrom**
Several of the Faithfulness metrics provide a plot as the results. In many cases, it can be nice to summarize the results with a single number. This can either be done through some mean aggregation or by calculating the auc of the plots produced by the metrics. In this PR, I have implemented a "return_auc" parameter for the PixelFlipping, RegionPerturbation, and Selectivity metrics, and a "return_aggregate" parameter for the SensitivityN metric.

### CODE REVIEW

File: quantus/metrics/faithfulness_metrics.py [hunk 1]
```diff
@@ -1414,13 +1414,17 @@ def __call__(
                 x_input = model.shape_input(x_perturbed, x.shape, channel_first=True)
                 y_pred_perturb = float(model.predict(x_input, **self.kwargs)[:, y])
                 preds.append(y_pred_perturb)
-
-            self.last_results.append(preds)
+            
+        self.last_results.append(preds)
```
Comment (annahedstroem) [inline quantus/metrics/faithfulness_metrics.py:1418]: line 1418 should have an extra indent?

File: quantus/metrics/faithfulness_metrics.py [hunk 2]
```diff
@@ -1714,6 +1718,11 @@ def __call__(
 
         return self.last_results
 
+    @property
```
Comment (annahedstroem) [inline quantus/metrics/faithfulness_metrics.py:1721]: super

File: quantus/metrics/faithfulness_metrics.py [hunk 3]
```diff
@@ -1967,6 +1978,11 @@ def __call__(
...
+    def get_auc_score(self):
+        """Calculate the area over the curve (AOC) score for several test samples."""
```
Comment (annahedstroem) [inline quantus/metrics/faithfulness_metrics.py:1983]: isn't area **under** the curve calculated or am I wrong=

File: quantus/metrics/faithfulness_metrics.py [hunk 4]
```diff
@@ -2018,6 +2036,7 @@ def __init__(self, *args, **kwargs):
             features_in_step (integer): The size of the step, default=1.
             max_steps_per_input (integer): The number of steps per input dimension, default=None.
             softmax (boolean): Indicates wheter to use softmax probabilities or logits in model prediction.
+            return_aggregate (boolean): Indicates whether an aggregated(mean) metric is returned, default=True.
```
Comment (annahedstroem) [inline quantus/metrics/faithfulness_metrics.py:2039]: great

File: tests/metrics/test_faithfulness_metrics.py [hunk 5]
```diff
@@ -411,7 +411,7 @@ def test_faithfulness_estimate(
                 "disable_warnings": True,
                 "display_progressbar": True,
             },
-            {"min": 0.0, "max": 1.0},
+            {"min": 0.0, "max": 80.0},
```
Comment (annahedstroem) [inline tests/metrics/test_faithfulness_metrics.py:414]: can we make it explicit in the text if we use softmax or not? so that min and max make sense?

**Review by annahedstroem — CHANGES_REQUESTED** [review]
Awesome!

2 small comments, please have a look :-)

File: quantus/metrics/faithfulness_metrics.py (hunk 1 above)
Comment (Wickstrom) [reply quantus/metrics/faithfulness_metrics.py]: Correct, good spot! Fixed it.

File: quantus/metrics/faithfulness_metrics.py (hunk 3 above)
Comment (Wickstrom) [reply quantus/metrics/faithfulness_metrics.py]: Right again, I have updated it.

**Comment by Wickstrom** [comment]
Good comments. I incorporated them in the last PR. Also, I added the option to return aggregates or not in the IROF score.

**Review by annahedstroem — COMMENTED** [review]
OK. Ready for merge.

**Comment by annahedstroem** [comment]
Ok there is a test failing

`FAILED tests/metrics/test_faithfulness_metrics.py::test_selectivity[load_mnist_model-load_mnist_images-params5-expected5]`

Can you check @Wickstrom?

**Comment by Wickstrom** [comment]
Sorry about this @annahedstroem. I sat the bound in of the tests a little to low but passed the test by chance locally. Fixed it now.

**Review by annahedstroem — APPROVED** [review]
